### PR TITLE
Implement server group/cluster providers

### PR DESCRIPTION
**Pull request by scotmoor**
Added functionality to retrieve cluseter and server group data from cache and pass back to the UI

### CODE REVIEW

**Comment by scotmoor** [comment]
@spinnaker/netflix-reviewers PTAL

**Comment by scotmoor** [comment]
@spinnake/netflix-reviewers could someone take a look at this? We trying to get this in for our demo next week

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/cache/Keys.groovy [hunk 1]
```diff
@@ -46,7 +46,12 @@ class Keys {
     }
   }
 
+
   static Map<String, String> parse(AzureCloudProvider azureCloudProvider, String key) {
+    parse(azureCloudProvider.id, key)
+  }
+
+  static Map<String, String> parse(String providerId, String key){
```
Comment (cfieber) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/cache/Keys.groovy]: you can probably ditch the providerId here and only handle items where the id segment of the key == `AzureCloudProvider.AZURE`

**Comment by cfieber** [comment]
LGTM

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/cache/Keys.groovy (hunk 1 above)
Comment (scotmoor) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/cache/Keys.groovy]: Yeah, there is some more refactoring coming in this area

**Comment by scotmoor** [comment]
Thanks @cfieber I just rebased.

**Pull request merged**